Sources/Widgets/Widgets3D/ShapeWidget/example/index.js

```diff
@@ -276,10 +277,52 @@ reader
       }
     );
 
+    const updateWidgetVisibility = (widget, slicePos, i, widgetIndex) => {
+      const widgetVisibility =
+        widget.getWidgetState().getPoint1Handle().getOrigin()[i] ===
+          slicePos[i] ||
+        !scene.widgetManager.getWidgets()[widgetIndex].getPoint1();
+      widget.setVisibility(widgetVisibility);
+    };
+
+    const updateWidgetsVisibility = (position) => {
+      for (let i = 0; i < 3; i++) {
+        if (position[i] !== 0) {
+          /* testing if the widget is on the slice and has been placed to modify visibility */
+          updateWidgetVisibility(widgets.rectangleWidget, position, i, 0);
+          updateWidgetVisibility(widgets.ellipseWidget, position, i, 1);
+          updateWidgetVisibility(widgets.circleWidget, position, i, 2);
+        }
+      }
+    };
+
```

Comment:
@ocrossi do I understand correctly that this is to only show the widget representation if it's on the slice it's been initially created?

Reply:
yes exactly